### PR TITLE
CI: Don't assume collect script exists

### LINKED ISSUE #206: CI: Don't assume collect script is available
PR #194 assumed that the data collection script will always be available. That (will be) true for a packaged system, but is not necessarily true in a CI environment; the runtime repo is cloned, but is not built meaning `kata-collect-data.sh` won't be installed unless the distro under test is `runtime`.

### PR DESCRIPTION
**Pull request by jodh-intel**
Only call the collect script if it exists.

Fixes #206.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: .ci/teardown.sh [hunk 1]
```diff
@@ -46,7 +50,7 @@ if [ "${log_copy_dest}" ]; then
 	journalctl --no-pager -u docker > "${docker_log_path}"
 	journalctl --no-pager -u kubelet > "${kubelet_log_path}"
 
-	sudo kata-collect-data.sh > "${collect_data_log_path}"
+	[ "${have_collect_script}" = yes ] && sudo $collect_script > "${collect_data_log_path}"
```
Comment (devimc) [inline .ci/teardown.sh]: s/yes/"yes" 

sudo -E PATH="$PATH" $collect_script

File: .ci/teardown.sh [hunk 2]
```diff
@@ -59,14 +63,17 @@ if [ "${log_copy_dest}" ]; then
...
-		"${collect_data_log_prefix}"
+	prefixes=""
+	prefixes+=" ${kata_runtime_log_prefix}"
+	prefixes+=" ${proxy_log_prefix}"
+	prefixes+=" ${shim_log_prefix}"
+	prefixes+=" ${crio_log_prefix}"
+	prefixes+=" ${docker_log_prefix}"
+	prefixes+=" ${kubelet_log_prefix}"
+
+	 [ "${have_collect_script}" = yes ] && prefixes+=" ${collect_data_log_prefix}"
```
Comment (devimc) [inline .ci/teardown.sh]: s/yes/"yes"

File: .ci/teardown.sh [hunk 3]
```diff
@@ -91,6 +98,9 @@ else
 	echo "Kubelet Log:"
 	journalctl --no-pager -u kubelet
 
-	echo "Kata Collect Data script output"
-	sudo kata-collect-data.sh
+	if [ "${have_collect_script}" = yes ]
```
Comment (devimc) [inline .ci/teardown.sh]: s/yes/"yes"

File: .ci/teardown.sh [hunk 4]
```diff
@@ -91,6 +98,9 @@ else
...
+	then
+		echo "Kata Collect Data script output"
+		sudo $collect_script
```
Comment (devimc) [inline .ci/teardown.sh]: sudo -E PATH="$PATH" $collect_script

**Comment by jodh-intel** [comment]
@devimc - good calls - branch updated!

File: .ci/teardown.sh (hunk 3 above)
Comment (devimc) [reply .ci/teardown.sh]: "yes"    :)

Comment (jodh-intel) [reply .ci/teardown.sh]: "doh!"

"fixed" :)

**Comment by GabyCT** [comment]
lgtm

**Pull request merged**